src/tls.rs

```diff
@@ -0,0 +1,138 @@
+use std::io::{BufReader, Read, Write};
+use std::sync::{Arc, Mutex};
+use std::{fs, io};
+
+use rustls::{self, Connection, NoClientAuth, ServerConfig, ServerConnection};
+
+/// TLS configuration
+#[derive(Clone)]
+pub struct TlsConfig {
+    /// Full path to the server certificate file.
+    pub server_cert: String,
+    /// Full path to the server key file.
+    pub server_cert_key: String,
+}
+
+fn make_config(config: &TlsConfig) -> Result<Arc<rustls::ServerConfig>, io::Error> {
+    let builder = ServerConfig::builder()
+        .with_safe_default_cipher_suites()
+        .with_safe_default_kx_groups()
+        .with_safe_default_protocol_versions()
+        .map_err(|_| {
+            io::Error::new(
+                io::ErrorKind::Other,
+                "inconsistent cipher-suites/versions specified",
+            )
+        })?
+        .with_client_cert_verifier(NoClientAuth::new());
+
+    let mut server_config = {
+        let certs = load_cert(&config.server_cert)?;
+        let privkey = load_private_key(&config.server_cert_key)?;
+
+        builder
+            .with_single_cert(certs, privkey)
+            .map_err(|_| io::Error::new(io::ErrorKind::Other, "bad certificates/private key"))?
+    };
+
+    server_config.key_log = Arc::new(rustls::KeyLogFile::new());
+
+    Ok(Arc::new(server_config))
+}
+
+fn load_cert(filename: &str) -> Result<Vec<rustls::Certificate>, io::Error> {
```

Comment:
This feels like something that shouldn't live in this library.

Reply:
I agree.  Similar to above, I was trying to avoid exposing the fact that we're using rustls.  But this code clearly smells of my particular use cases, so I'll update this.